### PR TITLE
Feat/ashify Battle Wordle

### PR DESCRIPTION
**Pull request by rellen**
Refactors Battle Wordle to use an ETS-backed resource for game state.
Changes are implemented as Ash changes, with Ash validations.

TODO: concurrency control - at the moment there's no guarantee at the UI level that the game state is up to date before a change is made.

### CODE REVIEW

**Review by zachdaniel — APPROVED** [review]

**Comment by zachdaniel** [comment]
Nicely done!

File: lib/playdate/games/battle_wordle/app_provider.ex [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  @impl true
+  def root_component, do: Playdate.Games.BattleWordle.UI.GameLive
+
+  @doc """
+  Start a Battle Wordle game server for an instance
+  """
+  @impl true
+  def start_link(instance) do
+    secret_word = Dictionary.get_secret()
+    BW.create_game()
```
Comment (mbuhot) [inline lib/playdate/games/battle_wordle/app_provider.ex:28]: Does this line need to be here? There's another call to `create_game!` below.

File: lib/playdate/games/battle_wordle/game.ex [hunk 2]
```diff
@@ -1,310 +1,115 @@
...
+      argument :player_name, :string, allow_nil?: false
+      argument :letter, :string, allow_nil?: false
 
-          updated_player = %Player{
-            updated_player
-            | current_guess: "",
-              guesses: [guess_result | guesses]
-          }
+      validate attribute_equals(:status, :started)
+      validate {Playdate.Games.BattleWordle.Game.ValidatePlayerExists, []}, expensive?: true
```
Comment (mbuhot) [inline lib/playdate/games/battle_wordle/game.ex:42]: `expensive?: true` TIL nice 👍

Comment (zachdaniel) [reply lib/playdate/games/battle_wordle/game.ex:42]: In 2.0 this is renamed to `before_action?: true` for clarity.

File: lib/playdate/games/battle_wordle/server.ex [hunk 3]
```diff
@@ -0,0 +1,51 @@
+defmodule Playdate.Games.BattleWordle.Server do
```
Comment (mbuhot) [inline lib/playdate/games/battle_wordle/server.ex:1]: Is this module still required?

File: lib/playdate/games/battle_wordle/ui/game_live.ex [hunk 4]
```diff
@@ -90,45 +89,52 @@ defmodule Playdate.Apps.BattleWordle.UI.GameLive do
...
   end
 
   defp player_rows(game_state, player) do
-    letter_type =
-      case Game.guess_valid?(game_state, player.current_guess) do
-        {:error, :invalid_guess} -> :invalid_word
-        _ -> :initial
-      end
+    letter_type = :initial
+    # case Game.guess_valid?(game_state, player.current_guess) do
```
Comment (mbuhot) [inline lib/playdate/games/battle_wordle/ui/game_live.ex:134]: Comments can be removed?

File: test/support/factories/app_factory.ex [hunk 5]
```diff
@@ -51,7 +51,8 @@ defmodule Playdate.Factories.AppFactory do
     %Playdate.Apps.AppUser{
       id: Ecto.UUID.generate(),
       session_id: Ecto.UUID.generate(),
-      user_id: Ecto.UUID.generate()
+      user_id: Ecto.UUID.generate(),
+      nickname: Faker.Person.first_name()
```
Comment (mbuhot) [inline test/support/factories/app_factory.ex:55]: 👍

File: lib/playdate/games/battle_wordle/ui/game_live.ex [hunk 6]
```diff
@@ -1,11 +1,11 @@
...
+defmodule Playdate.Games.BattleWordle.UI.GameLive do
   use PlaydateWeb, :live_view
   use Phoenix.Component
 
-  alias Playdate.Apps.BattleWordle.Game
-  alias Playdate.Apps.BattleWordle.GameServer
-  alias Playdate.Apps.BattleWordle.UI.Keyboard
   alias Playdate.Games.AppHelpers
+
+  alias Playdate.Games.BattleWordle, as: BW
```
Comment (Theosaurus-Rex) [inline lib/playdate/games/battle_wordle/ui/game_live.ex:7]: Probably nitpicking but are we able to just alias this as BattleWordle for clarity?

File: lib/playdate/games/battle_wordle/server.ex (hunk 3 above)
Comment (rellen) [reply lib/playdate/games/battle_wordle/server.ex:1]: No, forgot to delete it

File: lib/playdate/games/battle_wordle/ui/game_live.ex (hunk 4 above)
Comment (rellen) [reply lib/playdate/games/battle_wordle/ui/game_live.ex:134]: We need a todo here (I forgot to do it).  We need a guess_valid? against a Game Ash record here to decide whether to render the row on the board as invalid (not a member of the allowed guess list)

File: lib/playdate/games/battle_wordle/app_provider.ex (hunk 1 above)
Comment (rellen) [reply lib/playdate/games/battle_wordle/app_provider.ex:28]: Nope.  Copy pasta error.

**Pull request merged**